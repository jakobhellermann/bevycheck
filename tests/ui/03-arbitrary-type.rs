#![allow(unused)]
use bevy::prelude::*;

struct Foo;

#[bevycheck::system]
fn system(_: Foo) {}

fn main() {
    system.system();
}
