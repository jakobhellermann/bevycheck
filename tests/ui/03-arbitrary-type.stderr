error[E0277]: the trait bound `Foo: SystemParam` is not satisfied
 --> tests/ui/03-arbitrary-type.rs:7:14
  |
7 | fn system(_: Foo) {}
  |              ^^^ the trait `SystemParam` is not implemented for `Foo`
  |
  = help: the following other types implement trait `SystemParam`:
            &'a Archetypes
            &'a Bundles
            &'a Entities
            &'a RemovedComponentEvents
            &'a bevy::ecs::component::Components
            &World
            ()
            (P0, P1)
          and $N others
note: required by a bound in `is_system_param`
 --> tests/ui/03-arbitrary-type.rs:6:1
  |
6 | #[bevycheck::system]
  | ^^^^^^^^^^^^^^^^^^^^ required by this bound in `is_system_param`
  = note: this error originates in the attribute macro `bevycheck::system` (in Nightly builds, run with -Z macro-backtrace for more info)
